src/drive/web/modules/drive/Toolbar/components/DocumentQualification.jsx

```diff
@@ -133,42 +143,109 @@ DocumentCategory.propTypes = {
  * its parent
  *
  */
+const filename_extension = '.jpg'
 class DocumentQualification extends Component {
-  state = {
-    selected: {
-      categoryLabel: null,
-      itemId: null
+  constructor(props) {
+    super(props)
+    const { categoryLabel = null, itemId = null } = props.initialSelected || {}
+    this.defaultFilename = `Scan_${new Date().toISOString().replace(/:/g, '-')}`
+    this.state = {
+      selected: { categoryLabel, itemId },
+      filename: this.defaultFilename,
+      shouldAutomaticalyRenameFile: true
     }
+    this.textInput = React.createRef()
   }
 
+  getFilenameFromCategory = (item, t) => {
+    const realItem = getItemById(item.itemId)
+    if (realItem) {
+      const name = t(`Scan.items.${realItem.label}`)
+      return `${name.replace(/ /g, '-')}_${new Date()
+        .toLocaleDateString()
+        .replace(/:/g, '-')
+        .replace(/\//g, '-')}`
+    } else {
+      return this.defaultFilename
+    }
+  }
   onSelect = item => {
-    this.setState({ selected: item })
+    const { t } = this.props
+    const { shouldAutomaticalyRenameFile } = this.state
+    let filename = null
+    if (shouldAutomaticalyRenameFile) {
+      filename = this.getFilenameFromCategory(item, t)
+    } else {
+      filename = this.state.filename
+    }
+    this.setState({ selected: item, filename })
+
     const { onQualified } = this.props
     if (onQualified) {
       const realItem = getItemById(item.itemId)
-      //We only call the callback if a "real" item is selected
-      //not if `Scan.categories.undefined` is
-      if (realItem) onQualified(realItem)
+      onQualified(
+        realItem ? realItem : undefined,
+        filename + filename_extension
```

Comment:
How can the file extension be hardcoded? 🤔

Reply:
OK I see why you have this question. 

I'll add a few `if(editFileName)` to better reflect the process and have a better understanding. Currently, this second argument is only used in the Scanner and not in the Edit